### ISSUE TITLE
Views UI modals do not advance correctly when adding multiple fields

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#1930]
When adding multiple fields to a view at the same time, the modals do not advance correctly from configuration for one field to the next. The following errors are thrown:

```
Warning: Parameter 3 to views_ui_build_form_state() expected to be a reference, value given in views_ui_ajax_form() (line 2871 of backdrop/core/modules/views_ui/views_ui.admin.inc).
Notice: Undefined index: form_key in views_ui_build_form_path() (line 2748 of backdrop/core/modules/views_ui/views_ui.admin.inc).
Notice: Undefined index: view in views_ui_build_form_path() (line 2750 of backdrop/core/modules/views_ui/views_ui.admin.inc).
Notice: Trying to get property of non-object in views_ui_build_form_path() (line 2750 of backdrop/core/modules/views_ui/views_ui.admin.inc).
Notice: Undefined index: form_key in views_ui_build_form_path() (line 2751 of backdrop/core/modules/views_ui/views_ui.admin.inc).
Notice: Undefined index: display_id in views_ui_build_form_path() (line 2751 of backdrop/core/modules/views_ui/views_ui.admin.inc).
```

**Comment by oadaeh** [comment]
This is still an issue.
Based on the errors, it looked like it might a simple enough problem to solve, but when I scanned the code, it looked more like some logic error in determining whether to make the call or not.

`    if (!empty($view->stack)) {
      $backdrop_add_js = $backdrop_add_js_original;
      $stack = $view->stack;
      $top = array_shift($stack);
      $top[0] = $js;
      $form_state = call_user_func_array('views_ui_build_form_state', $top);
`

Unfortunately, I have a days long list to complete today, and I'm already 4 hours working on things I hadn't planned on, so here's where I stop for now. :(

**Comment by bradbulger** [comment]
If I pull the values of the $top array in that code into explicit variables and use them in the call to views_ui_build_form_state() the whole cascade of errors goes away. Though I'd expect that the $view and $display_id variables are the same. Doing comparisons, that appears to be the case.

**Comment by klonos** [comment]
Would be great to have steps to reproduce this. Here's what I've tried:

1. Start creating a view, leaving all defaults except for "Display format" 
2. Set "Display format" to "Table" (which makes so that fields are used)
3. Click "Continue & configure" -> "Content: Title (Title)" is added in the "Fields" section 👍 
4. Click "Add" in the "Fields" section, and add the "Content: Body (Body)" and "Content: Image (Image)" fields at the same time.
5. Progress through the steps of configuring the fields added in the previous step.

I got no errors 🤷‍♂ 

...anyway, this seems to be https://www.drupal.org/project/views/issues/2810431, which is part of #3853.

Can you guys see if https://github.com/backdrop/backdrop/pull/3056 fixes this? Thanks.

**Comment by bradbulger** [comment]
Yes, I could recreate the problem without that change on a test view, then applied the change and the problem stopped.

**Comment by bradbulger** [comment]
I also saw that some fields didn't trigger the problem and others did, though I didn't try the steps you outlined. I made a new view of all content with a page showing an unformatted list of fields, and added several fields that were common to everything like body, comment count, etc.

**Comment by klonos** [comment]
> Yes, I could recreate the problem without that change on a test view, then applied the change and the problem stopped.

Great! ...based on this comment, I've updated the status of this issue to WFM.

**Comment by bradbulger** [comment]
Looks like this was fixed by #4328

**Comment by quicksketch** [comment]
Looks like this was fixed back in 1.15.1, per #4312 

The PR over in https://github.com/backdrop/backdrop/pull/3056 has also been merged, if that was responsible for the problem.

Either way, it appears this has been fixed. Please reopen if that is not the case. Thanks @bradbulger for your investigation!
